Shake.hs

```diff
@@ -142,34 +140,29 @@ main = do
         need [dropExtension (un out) -<.> "hs"]
         compileBanner out
 
-      _ <- addOracle $ \(GhcPkg _) -> do
-        Stdout out <- command [] "ghc-pkg" ["dump"]
-        return $ words out
-
       when (m /= Build) (action $ runWeb m imgExt)
 
       return ()
 
 compileImg :: Bool -> FilePath -> Action ()
 compileImg isThumb outPath = do
-    systemCwdNorm "web/gallery" (obj "web/gallery/BuildGallery.hs.exe")
+    runExe [] "BuildGallery"
       ( (if isThumb then [ "--thumb", "200" ] else [])
-        ++ [(takeBaseName . takeBaseName) outPath, "../.." </> outPath]
+        ++ [(takeBaseName . takeBaseName) outPath, outPath, "web/gallery"]
```

Comment:
Here I changed the BuildGallery script to accept an additional path argument. However, I now realize that I could just add a `Cwd "web/gallery"` argument to my call to `runExe`, and then the old version of `BuildGallery` would work.